sql: do not show FAMILY if there is only one on SHOW CREATE TABLE

otan opened the pull request:
Release note (sql change): SHOW CREATE TABLE no longer shows the FAMILY
clause if there is only the PRIMARY family clause.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76285)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm if we get PM signoff

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76285#-:-MvUL9u49NTZh6JtyKDT:bfle0nj)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[pkg/sql/show_create_clauses.go, line 402 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76285#-MvUL0r33DgLeNlZtniX:-MvUL0r33DgLeNlZtniY:b-u34vom) ([raw file](https://github.com/cockroachdb/cockroach/blob/65deaed84a676a1e0c8c265f519d746971cd4dea/pkg/sql/show_create_clauses.go#L402)):*
> ```Go
> 	// it is named "primary".
> 	families := desc.GetFamilies()
> 	if len(families) == 1 && families[0].Name == "primary" {
> ```

i think there's a const we can use instead of hard-coding the string


<!-- Sent from Reviewable.io -->

adityamaru commented:
Bulk changes LGTM!

otan committed:
sql: do not show FAMILY if there is only one on SHOW CREATE TABLE

Release note (sql change): SHOW CREATE TABLE no longer shows the FAMILY
clause if there is only the PRIMARY family clause.

otan commented:
ok has been given

bors r=rafiss
